src/Lightning/NoiseProtocol/OldHkdf.cs

```diff
@@ -0,0 +1,97 @@
+using System;
+using System.Diagnostics;
+
+namespace NoiseProtocol
+{
+   public class OldHkdf : IHkdf, IDisposable
```

Comment:
Please add comments about why this is still used and what we need to do in the future to fix it (and then delete this class)